### PR TITLE
Deflake mach-launch test with increased timeout

### PR DESCRIPTION
**Pull request by raubitsj**
Description
-----------
Attempt to fix:
- https://app.circleci.com/pipelines/github/wandb/wandb/14230/workflows/10c50442-281c-4962-b9a6-cf5518aa7a20/jobs/243430
- https://app.circleci.com/pipelines/github/wandb/wandb/14229/workflows/fba1d6ff-bc97-49b1-bf54-724a2d2cb62f/jobs/243380

Testing
-------
just circleci

### CODE REVIEW

File: tests/unit_tests/tests_launch/test_launch.py [hunk 1]
```diff
@@ -1157,6 +1157,7 @@ def test_run_in_launch_context_with_malformed_env_vars(
         assert "Malformed WANDB_ARTIFACTS, using original artifacts" in err
 
 
+@pytest.mark.timeout(240)
```
Comment (kptkin) [inline tests/unit_tests/tests_launch/test_launch.py:1160]: wow... this test takes a long time... maybe we should move to nightly?

File: tox.ini [hunk 2]
```diff
@@ -40,7 +40,7 @@ deps =
     {[unitbase]deps}
     -r{toxinidir}/requirements_dev.txt
 install_command =
-    py{36,37,38,39,launch}: pip install --extra-index-url https://download.pytorch.org/whl/cpu {opts} {packages}
+    py{36,37,38,39,launch}: pip install --timeout 600 --extra-index-url https://download.pytorch.org/whl/cpu {opts} {packages}
```
Comment (kptkin) [inline tox.ini:43]: why install takes so much time???

File: tests/unit_tests/tests_launch/test_launch.py (hunk 1 above)
Comment (raubitsj) [reply tests/unit_tests/tests_launch/test_launch.py:1160]: all i know is that it takes more than 60 seconds occasionally-- i just chose 240 be safe and that is what is done for other tests.

but yeah, we should move things around to nightly as needed.. but note that we only have yea in nightly right now, not pytest

File: tox.ini (hunk 2 above)
Comment (raubitsj) [reply tox.ini:43]: TCP socket timeouts, see the PR description for the two failures that these two fixes address.
we should considering looking into creating a local pypi cache eventually

Comment (kptkin) [reply tox.ini:43]: got it! Thanks!

File: tests/unit_tests/tests_launch/test_launch.py (hunk 1 above)
Comment (kptkin) [reply tests/unit_tests/tests_launch/test_launch.py:1160]: yeah... we probably don't want tests so long gating PRs...

**Review by kptkin — APPROVED** [review]

**Pull request merged**